src/media/media.ts

```diff
@@ -605,13 +612,37 @@ export class SdkMedia extends (EventEmitter as { new(): StrictEventEmitter<Event
    * @returns the found device or `undefined` if the
    *  device could not be found.
    */
-  findCachedDeviceByTrackLabel (track?: MediaStreamTrack): MediaDeviceInfo | undefined {
+  findCachedDeviceByTrackLabelAndKind (track?: MediaStreamTrack): MediaDeviceInfo | undefined {
     if (!track) return;
     return this.getDevices().find(d =>
-      d.label === track.label && `${d.kind.substr(0, 5)}` === track.kind
+      d.label === track.label && `${d.kind.substring(0, 5)}` === track.kind.substring(0, 5)
```

Comment:
what's with the substrings?

Reply:
So I didn't dig down too deep, but sometimes the `kind` on the track is `videoinput` and sometimes it is just `video`. I don't know why. So we just substring it to always be sure.